src/btree/bt_sync.c

```diff
@@ -40,6 +40,9 @@ __sync_checkpoint_can_skip(WT_SESSION_IMPL *session, WT_REF *ref)
      *     skip them,
      * 5. there's already an address for every disk block involved.
      */
+    // TODO - we may want to disable during shutdown and recovery as well.
+    if (F_ISSET(S2C(session), WT_CONN_RTS_RUN))
+        return (false);
     if (WT_IS_HS(session->dhandle))
```

Comment:
Shall we add the recovery and shutdown conditions also?

Reply:
We haven't proven that we are leaving btrees dirty during recovery or shutdown. I suggest to prove it first that this is happening or are you convinced it is a possibility?